### PR TITLE
DM-11392: Add display tutorial as Part3 of Getting Started

### PR DESCRIPTION
**Pull request by jonathansick**
This tutorial covers the display framework for visualizing source images and catalogs. The tutorial is part of the Getting Started series to introduce readers to the software.

I've inserted the display tutorial after the processCcd.py tutorial so we can view images and source catalogs from processCcd.py before going on to coaddition.

The commit also updates the tutorial numbering sequence.

Draft: https://pipelines.lsst.io/v/DM-11392/getting-started/display.html

### CODE REVIEW

**Comment by jonathansick** [comment]
I think the tutorial needs `setup display_ds9`.

**Comment by laurenam** [comment]
I was just looking at it (finally).  I get
`$ eups list -s display_ds9
   13.0-2-g6174d4c+49   w_2017_37 current setup`
with setup lsst_distrib

It does need a fix in `for maskName, maskBit in mask.getMaskPlaneDict().items():` as you haven't defined mask.  `mask = calexp.getMask()` should do it.

File: getting-started/display.rst [hunk 1]
```diff
@@ -0,0 +1,361 @@
...
+Interpreting displayed mask colors
+==================================
+
+The display framework renders each plane of the mask in a different color (*plane* being a different bit in the mask).
+To interpret these colors you can get a dictionary of mask planes from the ``calexp`` and query the display for the colors it rendered each mask plane with.
+Run:
+
+.. code-block:: python
+
+   for maskName, maskBit in mask.getMaskPlaneDict().items():
```
Comment (laurenam) [inline getting-started/display.rst]: Need to define **mask**, e.g. `mask = calexp.getMask()`

File: getting-started/display.rst [hunk 2]
```diff
@@ -0,0 +1,361 @@
...
+   UNMASKEDNAN: None
+   NO_DATA: orange
+   BAD: red
+   EDGE: yellow
+   SUSPECT: yellow
+   NOT_DEBLENDED: None
+   CR: magenta
+   SAT: green
+
+Footprints of detected sources are rendered in blue and the saturated cores of bright stars are drawn in green.
```
Comment (laurenam) [inline getting-started/display.rst]: Should "Footprints" link to a definition, e.g.
https://confluence.lsstcorp.org/display/LSWUG/Astro+Glossary#AstroGlossary-Footprint
?

File: getting-started/display.rst [hunk 3]
```diff
@@ -0,0 +1,361 @@
...
+To make masked regions semi-transparent, so that underlying image features are visible, try:
+
+.. code-block:: python
+
+   display.setMaskTransparency(60)
+
+The ``setMaskTransparency`` method's argument can range from ``0`` (fully opaque) to ``100`` (fully transparent).
+
+You can also control the colorbar scaling algorithm with the display's ``scale`` method.
+Try an ``asinh`` stretch with the ``zscale`` algorithm for automatically selecting the white and black points:
```
Comment (laurenam) [inline getting-started/display.rst]: I don't think "points" is the right descriptor here (thresholds/extrema/??).

File: getting-started/display.rst [hunk 4]
```diff
@@ -0,0 +1,361 @@
...
+
+.. code-block:: python
+
+   print(src.getSchema())
+
+To get just the names of columns, run:
+
+.. code-block:: python
+
+   print(src.getSchema().getNames())
```
Comment (laurenam) [inline getting-started/display.rst]: `src.schema` (or `src.getSchema()`) also provides extra info (doc string, type, etc.)

Or, for a single element: `src.schema.find("calib_psfUsed")`

File: getting-started/display.rst [hunk 5]
```diff
@@ -0,0 +1,361 @@
...
+
+      src.asAstropy()
+
+   The returned Astropy Table is a view, not a copy, so it doesn't waste memory.
+
+Plotting sources on the display
+===============================
+
+Now let's overplot sources from the ``src`` table onto the image display using the ``Display``\ ’s ``dot`` method for plotting markers.
+``Display.dot`` plots markers individually, so we iterate iterate over rows in the ``SourceTable`` as usual.
```
Comment (laurenam) [inline getting-started/display.rst]: "iterate iterate" -> "iterate"

File: getting-started/display.rst [hunk 6]
```diff
@@ -0,0 +1,361 @@
...
+   with display.Buffering():
+       for s in src:
+           display.dot("o", s.getX(), s.getY(), size=10, ctype='orange')
+
+Now orange circles should appear in the DS9 window over every detected source.
+
+.. note::
+
+   Notice the ``getX`` and ``getY`` methods for getting the (x,y) centroid of each source.
+   These methods are shortcuts, using the table's *slot* system, to preferred columns for a source's centroid.
```
Comment (laurenam) [inline getting-started/display.rst]: Remove (or reword) ", to preferred columns for a source's centroid."

File: getting-started/display.rst [hunk 7]
```diff
@@ -0,0 +1,361 @@
...
+
+   display.erase()
+
+Selecting PSF-fitting sources to plot on the display
+====================================================
+
+Now lets be more purposeful and use the display to understand what sources were used for PSF measurement.
+
+The ``src`` table's ``calib_psfUsed`` column describes whether the source was used for PSF measurement.
+Since columns are Numpy arrays we can iterate over rows where ``src['calib_psfUsed'] == True`` with Numpy's boolean array indexing:
```
Comment (laurenam) [inline getting-started/display.rst]: Don't actually need `== True`.

File: getting-started/display.rst [hunk 8]
```diff
@@ -0,0 +1,361 @@
...
+
+   Notice the ``getX`` and ``getY`` methods for getting the (x,y) centroid of each source.
+   These methods are shortcuts, using the table's *slot* system, to preferred columns for a source's centroid.
+   Because the the ``src`` catalog contains measurements from several measurement plugins, slots are a way of easily using the pre-configured best measurements of a source.
+
+Clearing markers
+================
+
+``Display.dot`` always adds new markers to the display.
+To clear the display of all markers at any point, use the ``erase`` method:
```
Comment (laurenam) [inline getting-started/display.rst]: Remove "at any point"?

File: getting-started/display.rst [hunk 9]
```diff
@@ -0,0 +1,361 @@
...
+                            (src['calib_psfUsed'] == False)]
+   with display.Buffering():
+       for s in rejectedPsfSources:
+           display.dot("+", s.getX(), s.getY(), size=10, ctype='green')
+
+Now all green plus (**+**) symbols on the display mark rejected PSF measurement sources.
+
+The display framework, as we've seen, is a useful facility for inspecting images and tables.
+This tutorial only covered the framework's basic functionality.
+Explore the display framework documentation to learn how to display multiple images at once, and to work with different display backends.
```
Comment (laurenam) [inline getting-started/display.rst]: Link to the "display framework documentation"?

File: getting-started/display.rst [hunk 10]
```diff
@@ -0,0 +1,361 @@
...
+
+   src = butler.get('src', dataId={'filter': 'HSC-R', 'visit': 903334, 'ccd': 23})
+
+This ``src`` dataset is a ``SourceTable``, which is a table object from the ``lsst.afw.table`` module.
+
+We'll explore ``SourceTable``\ s more in a later tutorial, but you can check its length with Python's `len` function:
+
+.. code-block:: python
+
+   print(len(src))
```
Comment (laurenam) [inline getting-started/display.rst]: Provide a ballpark number to expect (with the caveat that it will be stack version dependent)?

**Pull request merged**